docs/dg/dev/miscellaneous-guides/strategy-resolver.md

```diff
@@ -0,0 +1,317 @@
+        );
+    }
+```
+
+### Using Context-Based Resolution in the Model
+
+When the model executes logic, it must explicitly specify the context when requesting a plug-in stack.
+
+```php
+class CheckoutWorkflow implements CheckoutWorkflowInterface
+{
+    /**
+     * @var \Spryker\Shared\Kernel\StrategyResolverInterface<list<\Spryker\Zed\CheckoutExtension\Dependency\Plugin\CheckoutPreConditionPluginInterface>>
+     */
+    protected $preConditionPluginStrategyResolver;
+
+    /**
+     * ...
+     * @param \Spryker\Shared\Kernel\StrategyResolverInterface<list<\Spryker\Zed\CheckoutExtension\Dependency\Plugin\CheckoutPreConditionPluginInterface>> $preConditionPluginStrategyResolver
+     * ...
+     */
+    public function __construct( ... StrategyResolverInterface $preConditionPluginStrategyResolver ...) {
+        ...
+        $this->preConditionPluginStrategyResolver = $preConditionPluginStrategyResolver;
+        ...
+    }
+
+    protected function checkPreConditions(QuoteTransfer $quoteTransfer, CheckoutResponseTransfer $checkoutResponse)
+    {
+        $isPassed = true;
+
+        $quoteProcessFlowName = $quoteTransfer->getQuoteProcessFlow()?->getNameOrFail(); // the context resolution depends on the needed business logic; this is just an example
+        $preConditionPlugins = $this->preConditionPluginStrategyResolver->get($quoteProcessFlowName);
+
+        foreach ($preConditionPlugins as $preConditionPlugin) {
+            $isPassed &= $preConditionPlugin->checkCondition($quoteTransfer, $checkoutResponse);
+        }
+
+        return (bool)$isPassed;
+    }
+}
+
+```
+
+### Enabling Lazy Plugin-Stack Resolution
+
+When multiple plugin-stacks are present with huge (10+) amount of plugins, performance may be an issue. For this reason,
+lazy loading is also introduced to the container system.
+The strategy resolver solves lazy loading in the background, ensuring that the correct plugin-stack is resolved only when requested.
+
+Note: The lazy loading can be used on demand even without strategy resolver, however fetching needs to be triggered on-demand.
+
+```php
+    $this->getProvidedDependency(CheckoutDependencyProvider::CHECKOUT_PRE_CONDITIONS, static::LOADING_LAZY);
+```
+
+### Debugging & Logging
+
+Everything remains typed, ensuring full compatibility with tools like Xdebug.
+The strategy resolver uses a generic template directive, allowing developers to inspect available plug-in stacks.
+
+### Frequently Asked Questions (FAQ)
+
+Q: Should I use this for all plugin-stacks?
+A: No, strategy design pattern comes with some complexity, so it is recommended to use only when business logic requires it.
+
+Q: Does this affect performance?
+A: Only if multiple plugin-stacks are resolved simultaneously. Use lazy loading to mitigate this.
+
+Q: What happens if a context is missing?
+A: By default, it fails. However, you can define a fallback mechanism.
+
+
+
+
+
+
+
+# Context-Based Dependency Resolution in Spryker
```

Comment:
Hey @gerner-spryker and @andriitserkovnyi 
The whole below part looks like a duplicate of the above. Not a full copy-paste though but I don't get how is it different from the above

If you decide to merge it with the above - there's additional Q&A point that @profuel has added, don't forget it